src/Tribe/Commerce/PayPal/Handler/IPN.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+class Tribe__Tickets__Commerce__PayPal__Handler__IPN {
+
+	/**
+	 * Set up hooks for IPN transaction communication
+	 *
+	 * @since TBD
+	 */
+	public function hook() {
+		add_action( 'template_redirect', array( $this, 'check_response' ) );
+	}
+
+	/**
+	 * Checks the request to see if payment data was communicated
+	 *
+	 * @since TBD
+	 */
+	public function check_response() {
+		if (
+			empty( $_POST )
+			|| ! isset( $_POST['txn_id'] )
+			|| ! isset( $_POST['payer_email'] )
+			|| ! $this->valid_transaction()
+		) {
+			return;
+		}
+
+		$paypal  = tribe( 'tickets.commerce.paypal' );
+		$gateway = tribe( 'tickets.commerce.paypal.gateway' );
```

Comment:
Any reasons to not make these object props? they're not going to change and you use `$gateway` in two functions in this object.

Reply:
Those objects aren't available for consumption during construction (or `hook()` execution) of this class. I could make a method that sets the property if it isn't set, but in the end that may be just as inefficient as executing the dependency injection function. I don't know the answer to that, though - perhaps @lucatume or @bordoni do?

Reply:
That's fine - just asking. Didn't flag as a required change 😉